src/internal/server/PrintLeftoverLabelController.go

```diff
@@ -22,6 +28,110 @@ type PrintLabelRequestBody struct {
 	Quantity  int    `json:"quantity"`
 }
 
+const FILE_PATH = "./tmp"
+
 func (c *PrintLeftoverLabelController) PrintLeftoverLabelHandler(w http.ResponseWriter, r *http.Request) {
-	// todo
+	/* -- FAIL FAST -- */
+
+	if r.Method != "POST" {
+		msg := "This endpoint only supports POST requests"
+		http.Error(w, msg, http.StatusBadRequest)
+		return
+	}
+
+	// ensure Content-Type "application/json"
+	ct := r.Header.Get("Content-Type")
+	if ct != "" {
+		mediaType := strings.ToLower(strings.TrimSpace(strings.Split(ct, ";")[0]))
+		if mediaType != "application/json" {
+			msg := "Content-Type header is not application/json. Received: " + mediaType
+			http.Error(w, msg, http.StatusUnsupportedMediaType)
+			return
+		}
+	}
+
+	if r.Body == nil {
+		msg := "Request body not provided"
+		http.Error(w, msg, http.StatusBadRequest)
+		return
+	}
+
+	/* -- PARSE AND VALIDATE BODY -- */
+
+	// read request body, ensuring the size is appropriate
+	r.Body = http.MaxBytesReader(w, r.Body, 128)
```

Comment:
What is an "appropriate" size? Can we extract this out to a constant and add a more descriptive comment here?